File: src/3state/states/ApplyState.ts
// ─── ApplyState — apply 子图状态 ─────────────────────────────────────────────
// LangGraph 子图必须与父图共享状态结构，故 ApplyState 在 MainState 基础上扩展

import { MessagesAnnotation, Annotation } from '@langchain/langgraph'
import type { ScoreTemplate } from '../../1common/types/shared.js'

export const ApplyState = Annotation.Root({
  ...MessagesAnnotation.spec,

  // ── 主图透传 ──────────────────────────────────────────────────
  intent:       Annotation<'consult' | 'apply' | 'insufficient'>({ reducer: (_, x) => x, default: () => 'apply' as const }),
  forcedIntent: Annotation<'consult' | 'apply' | null>({ reducer: (_, x) => x, default: () => null }),
  missingInfo:  Annotation<string[]>({ reducer: (_, x) => x, default: () => [] as string[] }),
  documentText: Annotation<string>({ reducer: (_, x) => x, default: () => '' }),

  // ── apply 专用 ────────────────────────────────────────────────
  // templates：加分模板列表（由 MCP 拉取）
  templates: Annotation<ScoreTemplate[]>({ reducer: (_, x) => x, default: () => [] as ScoreTemplate[] }),

  // policyContext：RAG 检索到的政策参考，fetchPolicyNode 填充
  policyContext: Annotation<string>({ reducer: (_, x) => x, default: () => '' }),

  // checkResults：LLM 匹配结果（JSON 字符串数组），analyzeMatchNode 输出
  checkResults: Annotation<string[]>({ reducer: (_, x) => x, default: () => [] as string[] }),
})

export type ApplyStateType = typeof ApplyState.State

File: src/state/apply_state.py
"""ApplyState - apply 子图状态 - 对应 TS: ApplyState.ts"""
from typing import Annotated, Literal, Sequence, TypedDict
from langgraph.graph import add_messages
from langchain_core.messages import BaseMessage


class ApplyState(TypedDict):
    """apply 子图状态 - 继承 MainState 并扩展"""
    
    messages: Annotated[Sequence[BaseMessage], add_messages]
    
    # 主图透传
    intent: Annotated[Literal["consult", "apply", "insufficient"], {"__root__": "apply"}]
    forced_intent: Annotated[Literal["consult", "apply"] | None, {"__root__": None}]
    missing_info: Annotated[list[str], {"__root__": []}]
    document_text: Annotated[str, {"__root__": ""}]
    
    # apply 专用
    templates: Annotated[list[dict], {"__root__": []}]
    
    # policyContext: RAG 检索到的政策参考
    policy_context: Annotated[str, {"__root__": ""}]
    
    # checkResults: LLM 匹配结果 (JSON 字符串数组)
    check_results: Annotated[list[str], {"__root__": []}]
